### ISSUE TITLE
RFC on new features

### ISSUE THREAD

**Issue by lexelby** [stesie/inkscape-embroidery#2]
I'm not ready for a PR yet, but I just wanted to invite you to check out what I've added to my fork.  I fixed the "preserve stacking order" option and added parsing for transforms so that ungrouping is not necessary.  I also added a method to override parameters for each path using extra attributes in the SVG (Edit -> Edit XML).

I'm going to use this to start adding underlay to my patterns to stabilize them and prevent puckering.  Eventually I'm hoping to add two new extensions: one to add empty-valued override parameters to every path in the document (to make it easier to override them) and one to automatically generate various styles of underlay for the selected paths.

I'd love to hear what you think! :)

**Comment by stesie** [comment]
Sorry for not answering earlier, anyways I really appreciate your work. I had a look at your code changes and it really looks promising to me. Next time I'm around in the Fablab nearby (where I have access to an embroidering machine) I'll definitely give it a try and report back :-)

**Comment by lexelby** [comment]
Yay! I'm looking forward to hearing how it works for you. :)

Since writing the comment above, I fixed several nasty bugs, added that params plugin, and, most excitingly, I added a variable-width satin option.  It's a little tricky to use, but hopefully the error messages guide you through it.  It helps solve issues with corners in the stroke-based satin and lets you do all sorts of nifty satin effects (teardrops, triangles, etc).  I'm super-excited about that one.

Next up, I want to improve the appearance of the fill stitching.  Right now it aligns each row of stitches randomly, and the eye naturally finds distracting patterns.  I'm hoping to make it stagger the stitches so that it looks more like a traditional tatami fill.  Not sure when I'll get to that, though.

**Comment by lexelby** [comment]
I've added a ton more to my repo, and at this I'm not sure a PR makes sense.  I've reworked major sections of the code including making fills look more like professional embroidery that I've seen.

Ultimately my goal is to beef this up into something close to a basic commercial embroidery tool.  There's just not enough out there for hobbyists to use for free and hack on.  I don't think I'll ever be able to match the ease of a dedicated digitizing program since inkscape plugins can't work interactively with the user, but I think I can get pretty close.

Anyway, I'm sure you're busy, but if you ever get a chance to try out my fork, I'd love to hear your feedback :)

BTW, I'm using a Brother SE400 (actually the protect version, but it's the same machine).  Maybe someday I'll graduate to a better machine :)
